Samples/OpenALSample/Program.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using CSCore;
+using CSCore.SoundOut;
+using CSCore.Codecs.WAV;
+using System.Threading;
+using System.Linq;
+
+namespace OpenALSample
+{
+	class MainClass
+	{
+		static ISoundOut _soundOut;
+		static IWaveSource _waveSource;
+		public static void Main (string[] args)
+		{
+			string path = "";
+			if (args.Length > 0) path = args [0];
+			else
+			{
+				Console.WriteLine ("Enter the wave file to play:");
+				path = Console.ReadLine();
+			}
+
+			_waveSource = new WaveFileReader (path);
+			_soundOut = new ALSoundOut ();
+			_soundOut.Initialize (_waveSource);
+
+			_soundOut.Play ();
+
+			while (true)
+			{
+				if (Console.KeyAvailable)
+				{
+					var key = Console.ReadKey ();
+					if (key.Key == ConsoleKey.Escape)
+						break;
+				}
+
+				var str = String.Format (@"{0:mm\:ss\.f}/{1:mm\:ss\.f}",
+					          TimeConverterFactory.Instance.GetTimeConverterForSource (_waveSource)
+					.ToTimeSpan (_waveSource.WaveFormat, _waveSource.Position),
```

Comment:
You may want to use the `Extensions.GetPosition`/`Extensions.GetLength` or `Extensions.GetTime` extension methods?